main.go

```diff
@@ -283,16 +283,19 @@ func main() {
 
 		diskCache := disk.New(c.Dir, int64(c.MaxSize)*1024*1024*1024, proxyCache)
 
+		metricsMdlw := httpmiddleware.New(httpmiddleware.Config{
+			Recorder: httpmetrics.NewRecorder(httpmetrics.Config{}),
+		})
```

Comment:
This doesn't seem to be used until a little further below. Could you move it to just before the `mux.Handle`  calls that first use `metricsMdlw`?

Reply:
done!